src/librustc/mir/repr.rs

```diff
@@ -32,6 +32,10 @@ pub struct Mir<'tcx> {
     /// that indexes into this vector.
     pub basic_blocks: Vec<BasicBlockData<'tcx>>,
 
+    /// List of lexical scopes; these are referenced by statements and
+    /// used (eventually) for debuginfo. Indexed by a `ScopeId`.
+    pub scopes: ScopeDataVec,
```

Comment:
I feel like it might be a good idea to have spans in the side table, along with scopes.